Task4_AWSCloud Formation.yaml

```diff
@@ -0,0 +1,235 @@
+    Default: 172.16.0.0/16
+    Description: CIDR block for the VPC
+  PublicSubnetCidr1:
+    Type: String
+    Default: 172.16.1.0/24
+    Description: CIDR block for Public Subnet 1 AZ1
+  PublicSubnetCidr2:
+    Type: String
+    Default: 172.16.2.0/24
+    Description: CIDR block for Public Subnet 2 AZ1
+  PrivateSubnetCidr1:
+    Type: String
+    Default: 172.16.3.0/24
+    Description: CIDR block for Private Subnet 1 AZ2
+  PrivateSubnetCidr2:
+    Type: String
+    Default: 172.16.4.0/24
+    Description: CIDR block for Private Subnet 2 AZ2
+
+Resources:
+ #VPC && InternetGateway.
+  #---- VPC Creation.
+  VPC:
+    Type: AWS::EC2::VPC
+    Properties:
+      CidrBlock: !Ref VpcCidr
+      EnableDnsSupport: true
+      EnableDnsHostnames: true
+      Tags:
+        - Key: Name
+          Value: DevVPC
+  #---- InternetGateway Creation.
+  InternetGateway:
+    Type: AWS::EC2::InternetGateway
+    Properties:
+      Tags:
+        - Key: Name
+          Value: InternetGatewayForDevVPC
+  #---- AttachGateway To VPC.
+  AttachGateway:
+    Type: AWS::EC2::VPCGatewayAttachment
+    Properties:
+      VpcId: !Ref VPC
+      InternetGatewayId: !Ref InternetGateway
+      DependsOn: 
+        - InternetGateway
+    # NAT Gateways and Elastic IPs
+  NatEip1:
+    Type: AWS::EC2::EIP
+    Properties:
+      Domain: vpc
+
+  NatEip2:
+    Type: AWS::EC2::EIP
+    Properties:
+      Domain: vpc
+
+  NatGateway1:
+    Type: AWS::EC2::NatGateway
+    Properties:
+      AllocationId: !GetAtt NatEip1.AllocationId
+      SubnetId: !Ref PublicSubnet1
+      Tags:
+        - Key: Name
+          Value: NatGateway1
+
+  NatGateway2:
+    Type: AWS::EC2::NatGateway
+    Properties:
+      AllocationId: !GetAtt NatEip2.AllocationId
+      SubnetId: !Ref PublicSubnet2
+      Tags:
+        - Key: Name
+          Value: NatGateway2
+ #RouteTables
+  #---- Public RouteTable Creation (Info:Any component will assassin here will send/receive requests to/from anywhere).
+  DefaultPublicRoute:
+    Type: AWS::EC2::RouteTable
+    Properties:
+      VpcId: !Ref VPC
+      DependsOn: 
+        - VPC
+      Tags:
+        - Key: Name
+          Value: DefaultPublicRoute
+  #-------- Attach Rout value to DefaultPublicRoute(Accept all 0.0.0.0/0 on Gateway).
+  PublicRoute:
+    Type: AWS::EC2::Route
+    Properties:
+      RouteTableId: !Ref DefaultPublicRoute
+      DestinationCidrBlock: 0.0.0.0/0
+      GatewayId: !Ref InternetGateway
+      DependsOn: 
+        - InternetGateway
+        - DefaultPublicRoute
+  #---- private RouteTable and routs Creation.
+  PrivateRouteTableSubNet1AZ1:
+    Type: AWS::EC2::RouteTable
+    Properties:
+      VpcId: !Ref VPC
+      Tags:
+        - Key: Name
+          Value: PrivateRouteTableSubNet1AZ1
+
+    PrivateRouteTableSubNet2AZ2:
+    Type: AWS::EC2::RouteTable
+    Properties:
+      VpcId: !Ref VPC
+      Tags:
+        - Key: Name
+          Value: PrivateRouteTableSubNet2AZ2
+
+  PrivateRoute1:
+    Type: AWS::EC2::Route

```

Comment:
You need to use a proper dependOn attribute

Reply:
It keep raise Error while creating the stack "extraneous key [DependsOn] is not permitted]"

Reply:
@enkhalifapro  Fixed "Move Depends on to its correct location"